changelog/6484.improvement.md

```diff
@@ -0,0 +1,4 @@
+Renamed `templates` to `responses` across Rasa code base.
```

Comment:
@yennycheung As it's not user facing I'd rather make it a `misc` changelog or even completely drop the changelog

Reply:
Wait I'm confused. In your suggestions below, we would want to add to the changelog, no?